src/cockpit/channels/filesystem.py

```diff
@@ -106,7 +106,10 @@ def do_yield_data(self, options: JsonObject) -> GeneratorChannel.DataGenerator:
                 if max_read_size is not None and buf.st_size > max_read_size:
                     raise ChannelError('too-large')
 
-                self.ready()
+                if binary:
```

Comment:
Probably you only want to do this if `stat.S_ISREG(buf.st_mode)` (and same for C)...